### PR TITLE
[Cocoa] Move contentSizeCategory to platform/

### PR DESCRIPTION
**Pull request by litherum**
#### c8a47f4117b4ed73f4dfd9126ab1157396ccd111
<pre>
[Cocoa] Move contentSizeCategory to platform/
<a href="https://bugs.webkit.org/show_bug.cgi?id=241360">https://bugs.webkit.org/show_bug.cgi?id=241360</a>

Reviewed by Cameron McCormack.

There are 2 reasons for moving it:
1. There are already layering violations where existing code in platform/ is calling into RenderTheme
       to get the contentSizeCategory.
2. For <a href="https://bugs.webkit.org/show_bug.cgi?id=237817">https://bugs.webkit.org/show_bug.cgi?id=237817</a>, I&apos;m going to be adding a new system font cache,
       and the most natural place to put it is in SystemFontDatabase in platform/. This new cache needs
       access to the contentSizeCategory, so contentSizeCategory needs to be in platform/.

This patch adds FontCacheCocoa.mm because the contentSizeCategory stuff is exposed via Objective-C, and
FontCacheCoreText.cpp is a C++ source.

* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/cocoa/FontCacheCocoa.mm: Copied from Source/WebCore/rendering/RenderThemeCocoa.h.
(WebCore::getUIContentSizeCategoryDidChangeNotificationName):
(WebCore::_contentSizeCategory):
(WebCore::contentSizeCategory):
(WebCore::setContentSizeCategory):
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::platformInit):
(WebCore::fontWithFamilySpecialCase):
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.h:
* Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
(WebCore::SystemFontDatabaseCoreText::createTextStyleFont):
* Source/WebCore/rendering/RenderThemeCocoa.h:
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::attachmentActionFont):
(WebCore::attachmentTitleFont):
(WebCore::attachmentDynamicTypeScaleFactor):
(WebCore::contentSizeCategoryDidChange): Deleted.
(WebCore::RenderThemeIOS::RenderThemeIOS): Deleted.
(WebCore::_contentSizeCategory): Deleted.
(WebCore::RenderThemeIOS::contentSizeCategory const): Deleted.
(WebCore::RenderThemeIOS::setContentSizeCategory): Deleted.
* Source/WebCore/rendering/RenderThemeMac.h:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::contentSizeCategory const): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::setContentSizeCategory):
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::contentSizeCategoryDidChange):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
* Tools/TestWebKitAPI/Tests/ios/TextStyleFontSize.mm:
(TEST):

Canonical link: <a href="https://commits.webkit.org/251365@main">https://commits.webkit.org/251365@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295351">https://svn.webkit.org/repository/webkit/trunk@295351</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Source/WebCore/platform/graphics/cocoa/FontCacheCocoa.mm [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+static String& _contentSizeCategory()
+{
+    static NeverDestroyed<String> _contentSizeCategory;
+    return _contentSizeCategory.get();
+}
+
+CFStringRef contentSizeCategory()
+{
+    if (!_contentSizeCategory().isNull())
+        return (__bridge CFStringRef)static_cast<NSString*>(_contentSizeCategory());
```
Comment (heycam) [inline Source/WebCore/platform/graphics/cocoa/FontCacheCocoa.mm]: Existing issue, but why do we cast the `_contentSizeCategory()` return value from `CFStringRef` to `NSString*` and then bridge cast it back to `CFStringRef` again?

File: Source/WebCore/platform/graphics/cocoa/FontCacheCocoa.mm [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+CFStringRef getUIContentSizeCategoryDidChangeNotificationName()
+{
+    return static_cast<CFStringRef>(PAL::get_UIKit_UIContentSizeCategoryDidChangeNotification());
+}
+
+static String& _contentSizeCategory()
+{
+    static NeverDestroyed<String> _contentSizeCategory;
+    return _contentSizeCategory.get();
+}
```
Comment (heycam) [inline Source/WebCore/platform/graphics/cocoa/FontCacheCocoa.mm]: Existing issue, but leading `_` does not feel like WebKit naming style.

File: Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp [hunk 3]
```diff
@@ -764,6 +763,10 @@ void FontCache::platformInit()
 {
     CFNotificationCenterAddObserver(CFNotificationCenterGetLocalCenter(), this, &fontCacheRegisteredFontsChangedNotificationCallback, kCTFontManagerRegisteredFontsChangedNotification, nullptr, CFNotificationSuspensionBehaviorDeliverImmediately);
 
+    CFNotificationCenterAddObserver(CFNotificationCenterGetLocalCenter(), this, &fontCacheRegisteredFontsChangedNotificationCallback, kAXSEnhanceTextLegibilityChangedNotification, nullptr, CFNotificationSuspensionBehaviorDeliverImmediately);
+
+    CFNotificationCenterAddObserver(CFNotificationCenterGetLocalCenter(), this, &fontCacheRegisteredFontsChangedNotificationCallback, getUIContentSizeCategoryDidChangeNotificationName(), 0, CFNotificationSuspensionBehaviorDeliverImmediately);
+
```
Comment (heycam) [inline Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp]: Does this belong in this patch? Seems unrelated to the code moving.

**Review by heycam — APPROVED** [review]

File: Source/WebCore/platform/graphics/cocoa/FontCacheCocoa.mm (hunk 1 above)
Comment (litherum) [reply Source/WebCore/platform/graphics/cocoa/FontCacheCocoa.mm]: ooooo this is the perfect place for a `bridge_cast`!!!

File: Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp (hunk 3 above)
Comment (litherum) [reply Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp]: Whoopsie!

Comment (litherum) [reply Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp]: The bottom one is necessary (and is a move) - see the previous implementation of `RenderThemeIOS::RenderThemeIOS()`.

The top one is a mistake.

**Comment by nt1m** [comment]
./platform/graphics/cocoa/FontCacheCoreText.cpp:766:135: error: use of undeclared identifier 'kAXSEnhanceTextLegibilityChangedNotification'
